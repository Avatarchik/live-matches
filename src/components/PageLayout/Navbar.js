import React from 'react';

const Navbar = () => (
  <nav>
    <ul>
      <li>
        <a href="instructions/index.html">Test instructions</a>
      </li>
    </ul>
  </nav>
);
export default Navbar;
