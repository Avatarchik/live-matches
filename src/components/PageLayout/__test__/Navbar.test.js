import React from 'react';

import { shallow } from 'enzyme';
import Navbar from '../Navbar';

describe('Test suite for Navbar Component', () => {
  // create a UI snapshot to indentify UI changes or design regressions

  it('render correctly text component', () => {
    const component = shallow(<Navbar />);
    expect(component).toMatchSnapshot();
    expect(component.find('ul').find('li')).toHaveLength(1);
  });
});
