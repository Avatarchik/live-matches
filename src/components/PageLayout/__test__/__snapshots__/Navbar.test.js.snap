// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Navbar Component render correctly text component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
          <a
            href="instructions/index.html"
          >
            Test instructions
          </a>
        </li>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <li>
          <a
            href="instructions/index.html"
          >
            Test instructions
          </a>
        </li>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="instructions/index.html"
          >
            Test instructions
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test instructions",
            "href": "instructions/index.html",
          },
          "ref": null,
          "rendered": "Test instructions",
          "type": "a",
        },
        "type": "li",
      },
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
            <a
              href="instructions/index.html"
            >
              Test instructions
            </a>
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <li>
            <a
              href="instructions/index.html"
            >
              Test instructions
            </a>
          </li>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="instructions/index.html"
            >
              Test instructions
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test instructions",
              "href": "instructions/index.html",
            },
            "ref": null,
            "rendered": "Test instructions",
            "type": "a",
          },
          "type": "li",
        },
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
